src/parse.rs

```diff
@@ -678,10 +678,24 @@ fn dlt_payload<T: NomByteOrder>(
     payload_length: u16,
     arg_cnt: u8,
     is_controll_msg: bool,
+    is_network_trace_msg: bool,
 ) -> IResult<&[u8], PayloadContent, DltParseError> {
     if verbose {
         match count(dlt_argument::<T>, arg_cnt as usize)(input) {
-            Ok((rest, arguments)) => Ok((rest, PayloadContent::Verbose(arguments))),
+            Ok((rest, arguments)) => {
+                if is_network_trace_msg {
+                    let slices = arguments
+                        .iter()
+                        .map(|i| match &i.value {
+                            Value::Raw(bytes) => bytes.clone(),
+                            _ => vec![], // TODO raise error?!
+                        })
+                        .collect();
+                    Ok((rest, PayloadContent::NetworkTrace(slices)))
+                } else {
+                    Ok((rest, PayloadContent::Verbose(arguments)))
+                }
+            }
```

Comment:
- Should this TODO be resolved before merging the PR?
- If can embedded into the match arm if you find it better :wink: 
```rust
            Ok((rest, arguments)) if is_network_trace_msg => {
                let slices = arguments
                    .iter()
                    .map(|i| match &i.value {
                        Value::Raw(bytes) => bytes.clone(),
                        _ => vec![], // TODO raise error?!
                    })
                    .collect();
                Ok((rest, PayloadContent::NetworkTrace(slices)))
            }
            Ok((rest, arguments)) => Ok((rest, PayloadContent::Verbose(arguments))),
```

Reply:
There's no spec on content of network-trace messages and expecting several raw-byte args is taken from observations on someip-traces. As returning empty array in case no bytes argument is present makes no sence, i changed the code to filter them out for now:

```
let slices = arguments
.iter()
.map(|i| match &i.value {
    Value::Raw(bytes) => bytes.clone(),
    _ => vec![],
})
.filter(|i| !i.is_empty())
.collect();
```